application-forum-ui/src/main/resources/ForumCode/Macros.xml

```diff
@@ -978,7 +982,9 @@ addthis:description="${escapetool.xml($addThisDesc)}"
       #set($topicTitle = $topicDoc.title)
       #set($rating = $services.ratings.getAverageRating($topicDoc).nbVotes)
       #set($topicComments = 0)
-      #set($topicAnswers = $services.query.hql(", BaseObject obj where doc.parent = :topic and doc.fullName = obj.name and obj.className = 'ForumCode.AnswerClass' and doc.fullName &lt;&gt; 'ForumCode.AnswerClass' order by doc.date desc").bindValue('topic', ${topicDoc.getFullName()}).execute())
+      ## Depending on XE version, we consider parent-child relationship for Pre NS or space-subspace for NS.
+      #set($hierarchyCondition = 'doc.parent=:parent or doc.space=:space')
+      #set($topicAnswers = $services.query.xwql("from doc.object(ForumCode.AnswerClass) as answer where $hierarchyCondition order by doc.date desc").bindValue('parent', $topicDoc.fullName).bindValue('space', $topicDoc.space).execute())
```

Comment:
Same. This is a clear sign that this type of code needs to go into a macro or something to avoid duplication like it is now.